shell/platform/linux/fl_renderer.cc

```diff
@@ -173,6 +239,137 @@ gboolean fl_renderer_present_layers(FlRenderer* self,
 
   fl_renderer_unblock_main_thread(self);
 
-  return FL_RENDERER_GET_CLASS(self)->present_layers(self, layers,
-                                                     layers_count);
+  if (!priv->view) {
+    return FALSE;
+  }
+
+  g_ptr_array_set_size(priv->textures, 0);
+  for (size_t i = 0; i < layers_count; ++i) {
+    const FlutterLayer* layer = layers[i];
+    switch (layer->type) {
+      case kFlutterLayerContentTypeBackingStore: {
+        const FlutterBackingStore* backing_store = layer->backing_store;
+        auto framebuffer = &backing_store->open_gl.framebuffer;
+        FlBackingStoreProvider* provider =
+            FL_BACKING_STORE_PROVIDER(framebuffer->user_data);
+        g_ptr_array_add(priv->textures, g_object_ref(provider));
+      } break;
+      case kFlutterLayerContentTypePlatformView: {
+        // Currently unsupported.
```

Comment:
Consider making this a TODO and tacking on the issue ID: https://github.com/flutter/flutter/issues/41724